### PR TITLE
Add data proxy error codes

### PR DESCRIPTION
**Pull request by millsp**
## Describe this PR

This PR adds data proxy error codes.

## Changes

Added a new data proxy section.

## What issue does this fix?

Comes along with https://github.com/prisma/prisma/pull/9893

## Any other relevant information

Can be merged once https://github.com/prisma/prisma/pull/9893 is released

### CODE REVIEW

File: content/400-reference/200-api-reference/250-error-reference.mdx [hunk 1]
```diff
@@ -396,3 +396,48 @@ Possible errors:
...
+#### <inlinecode>P5001</inlinecode>
+
+"This request must be retried"
+
+#### <inlinecode>P5002</inlinecode>
+
+The datasource is an invalid one:
+
+* "Could not parse URL of the datasource"
+* "Datasource URL should use prisma:// protocol"
```
Comment (tanberry) [inline content/400-reference/200-api-reference/250-error-reference.mdx:415]: Is the "protocol" part of this error message a variable? Do we mean to have the space before "protocol"?

File: content/400-reference/200-api-reference/250-error-reference.mdx [hunk 2]
```diff
@@ -396,3 +396,48 @@ Possible errors:
...
+
+"Requested resource does not exist"
+
+#### <inlinecode>P5004</inlinecode>
+
+The feature is not yet implemented:
+
+* "beforeExit event is not supported yet"
+* "Interactive transactions are not supported yet"
+
```
Comment (tanberry) [inline content/400-reference/200-api-reference/250-error-reference.mdx:428]: This is a tiny nit, feel free to ignore, but "yet" should precede "supported". ("beforeExit event is not yet supported")

File: content/400-reference/200-api-reference/250-error-reference.mdx [hunk 3]
```diff
@@ -396,3 +396,48 @@ Possible errors:
...
+#### <inlinecode>P5005</inlinecode>
+
+"Could not find the schema"
+
```
Comment (tanberry) [inline content/400-reference/200-api-reference/250-error-reference.mdx:432]: The Prisma schema? Do we want to provide guidance on what to do?

**Review by tanberry — APPROVED** [review]
Only a few nits... I'm assuming the error strings are defined somewhere and wold need to be edited as well, if we edit these?

**Comment by millsp** [comment]
I'm going to close this PR as we'll add this piece elsewhere (in the data proxy docs). Thanks for the suggestions, I'll carry them over.

**Pull request closed without merging**